File: qc-domain/src/main/java/com/eager/questioncloud/domain/question/repository/QuestionRepository.java
package com.eager.questioncloud.domain.question.repository;

import com.eager.questioncloud.common.PagingInformation;
import com.eager.questioncloud.domain.question.common.QuestionFilter;
import com.eager.questioncloud.domain.question.dto.QuestionInformation;
import com.eager.questioncloud.domain.question.model.Question;
import java.util.List;

public interface QuestionRepository {
    int countByQuestionFilter(QuestionFilter questionFilter);

    List<QuestionInformation> getQuestionInformation(QuestionFilter questionFilter);

    QuestionInformation getQuestionInformation(Long questionId, Long userId);

    List<Question> getQuestionsByQuestionIds(List<Long> questionIds);

    Boolean isAvailable(Long questionId);

    Question findByQuestionIdAndCreatorId(Long questionId, Long creatorId);

    Question get(Long questionId);

    Question save(Question question);

    List<QuestionInformation> findByCreatorIdWithPaging(Long creatorId, PagingInformation pagingInformation);

    int countByCreatorId(Long creatorId);
}

File: qc-core/qc-question/qc-question-core/src/main/kotlin/com/eager/questioncloud/question/repository/QuestionRepository.kt
package com.eager.questioncloud.question.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.question.common.QuestionFilter
import com.eager.questioncloud.question.domain.Question
import com.eager.questioncloud.question.dto.QuestionInformation

interface QuestionRepository {
    fun countByQuestionFilter(questionFilter: QuestionFilter): Int
    
    fun getQuestionInformation(
        questionFilter: QuestionFilter,
        pagingInformation: PagingInformation
    ): List<QuestionInformation>
    
    fun getQuestionInformation(questionId: Long): QuestionInformation // TODO Del userId Param
    
    fun getQuestionsByQuestionIds(questionIds: List<Long>): List<Question>
    
    fun isAvailable(questionId: Long): Boolean
    
    fun findByQuestionIdAndCreatorId(questionId: Long, creatorId: Long): Question
    
    fun get(questionId: Long): Question
    
    fun save(question: Question): Question
    
    fun getQuestionInformationByCreatorIdWithPaging(
        creatorId: Long,
        pagingInformation: PagingInformation
    ): List<QuestionInformation>
    
    fun findByCreatorId(creatorId: Long): List<Question>
    
    fun findByQuestionIdIn(questionIds: List<Long>): List<QuestionInformation> // TODO Refactor Fun Name
    
    fun countByCreatorId(creatorId: Long): Int
    
    fun increaseQuestionCount(questionId: Long)
    
    fun deleteAllInBatch()
}
